src/devices/Tcs3472x/Tcs3472x.cs

```diff
@@ -0,0 +1,227 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Drawing;
+using System.Threading;
+
+namespace Iot.Device.Tcs3472x
+{
+    public class Tcs3472xSensor : IDisposable
+    {
+        public const byte DefaultI2cAddress = 0x29;
+        private readonly I2cDevice _i2cDevice;
+        private byte _integrationTimeByte;
+        private double _integrationTime;
+        private bool _isLongTime;
+        private bool _autoDisposable;
+        private Gain _gain;
+
+        /// <summary>
+        /// Set/Get the time to wait for the sensor to read the data
+        /// Minimum time is 0.0024 s
+        /// Maximum time is 7.4 s
+        /// Be aware that it is not a linear function
+        /// </summary>
+        public double IntegrationTime
+        {
+            get { return _integrationTime; }
+            set
+            {
+                _integrationTime = value;
+                SetIntegrationTime(_integrationTime);
+            }
+        }
+
+        /// <summary>
+        /// Set/Get the gain
+        /// </summary>
+        public Gain Gain
+        {
+            get { return _gain; }
+            set
+            {
+                _gain = value;
+                _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.CONTROL), (byte)_gain });
```

Comment:
1. feels like the command bit should not be in the register and rather some const in this class (the `[Flags]` on the Register is tiny bit confusing)
2. you can use non-allocating pattern here

Reply:
Yes, I wrote a function for it.